analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -463,8 +468,18 @@
 
                     // For now, we ignore SDKs like the Dart SDK and the Flutter SDK in the analyzer.
                     when {
+                        pkgInfoFromLockFile["source"].textValueOrEmpty() == "path" -> {
+                            rawName = packageName
+                            vcs = VcsInfo(
+                                type = VcsType.UNKNOWN,
+                                url = "",
+                                revision = "",
+                                path = pkgInfoFromLockFile["description"]["path"].toString()
```

Comment:
I'm not sure about this. Having only the `path` in the `VcsInfo` might not be correct, as the "path" dependency might point to a location on the file system not managed via VCS.
Is the Downloader able to "download" sources for a package with a `VcsInfo` like this?

Reply:
> Is the Downloader able to "download" sources for a package with a `VcsInfo` like this?

No 😀 

My proposal would be to
1. only support the case when the package directory is below the project directory, otherwise log a warning
2. derive the `VcsInfo` from the project like: `VersionControlSystem.forDirectory(workingDir).getInfo()`
3. amend that VcsInfo with the path from `pkgInfoFromLockFile["description"]["path"]` (relative to the project root)

@bennati would that cover your use-case? I.e. where are the paths to the dependencies located in your case?

Reply:
> What do you mean with 3? the path should be relative to the project directory, instead of absolute?

Yes. Or phrased differently, a `VcsInfo`'s path must be relative to the repository root. Kotlin has a `relativeTo` extension function on `File`s that might come handy here.

> the path in a VcsInfo cannot be amended because it is a `val`, or am I doing something wrong?

The way to do this is via a `data class`' built-in `copy` function: `vcsInfo.copy(path = pkgInfoFromLockFile["description"]["path"])`.

Reply:
done